packages/core/src/commands/network-generate.ts

```diff
@@ -119,7 +119,11 @@ export class Command extends Commands.Command {
                 } as prompts.PromptObject<string>),
         );
 
-        if (Object.keys(flagsDefinition).find(flagName => !response[flagName])) {
+        // TODO: check this fix is acceptable
+        // the distribute flag is a boolean in the pre-existing tests
+        // and it is defined as a number in this.generateCryptoGenesisBlock()
```

Comment:
The value should be treated as a boolean, must've also been a mistake when copying over some of the code from 2.6

Reply:
So is the fix here acceptable?

The problem is that, under the previous implementation, a value intentionally set to false here would have triggered the error (similarly a number intentionally set to 0 would trigger the error).

Reply:
I think it's fine for now. I'll probably revisit it when I do some other refactors and cleanups around that command.